src/Draco.Compiler/Internal/Symbols/Metadata/MetadataTypeSymbol.cs

```diff
@@ -92,6 +100,34 @@ private ImmutableArray<Symbol> BuildMembers()
             result.Add(methodSymbol);
         }
 
+        // Fields
+        foreach (var fieldHandle in this.typeDefinition.GetFields())
+        {
+            var fieldDef = this.MetadataReader.GetFieldDefinition(fieldHandle);
+            // Skip special name
+            if (fieldDef.Attributes.HasFlag(FieldAttributes.SpecialName)) continue;
+            // Skip non-public
+            if (!fieldDef.Attributes.HasFlag(FieldAttributes.Public)) continue;
+            // Add it
+            var fieldSym = new MetadataFieldSymbol(
+                containingSymbol: this,
+                fieldDefinition: fieldDef);
+            result.Add(fieldSym);
+        }
+
+        var defaultName = MetadataSymbol.GetDefaultMemberAttributeName(this.typeDefinition, this.DeclaringCompilation!, this.MetadataReader);
+
+        // Properties
+        foreach (var propHandle in this.typeDefinition.GetProperties())
+        {
+            var propDef = this.MetadataReader.GetPropertyDefinition(propHandle);
+            var propSym = new MetadataPropertySymbol(
+                containingSymbol: this,
+                propertyDefinition: propDef,
+                defaultMemberName: defaultName);
+            if (propSym.Visibility == Api.Semantics.Visibility.Public) result.Add(propSym);
```

Comment:
Any reason we don't do the "continue pattern" with visibility like the above? Do properties have no direct visibility, or is it easier to just reuse our logic with the visibility determination?